quilt/tools/build.py

```diff
@@ -144,29 +172,49 @@ def _callback(count):
 
     return df
 
-def build_package(username, package, yaml_path):
+def build_package(username, package, yaml_path, checks_path=None, dryrun=False, env='default'):
     """
     Builds a package from a given Yaml file and installs it locally.
 
     Returns the name of the package.
     """
+    def find(key, value):
+        for k, v in value.items():
+            if k == key:
+                yield v
+            elif isinstance(v, dict):
+                for result in find(key, v):
+                    yield result
+            elif isinstance(v, list):
+                for d in v:
+                    for result in find(key, d):
+                        yield result
     build_dir = os.path.dirname(yaml_path)
     fd = open(yaml_path)
     docs = yaml.load_all(fd)
-    data = next(docs, None) # leave other dicts in the generator
-    if not isinstance(data, dict):
-        raise BuildException("Unable to parse YAML: %s" % yaml_path)
-
-    build_package_from_contents(username, package, build_dir, data)
-
-def build_package_from_contents(username, package, build_dir, data):
-    """
-    Builds package according to YAML node
-    """
-    contents = data.get('contents', {})
+    build_data = next(docs, None) # leave other dicts in the generator
+    if not isinstance(build_data, dict):
+        raise BuildException("Unable to parse build YAML: %s" % yaml_path)
+    if len(list(find('checks', build_data['contents']))) > 0 and 'checks' not in build_data:
```

Comment:
NOCHG: though I know it's more pythonic, it's harder for multi-language programmers to read so I like to be explicit.  ok?

Reply:
I think we should actually use the Python conventions:
- Doesn't make sense to optimize for people who are not familiar with the language
- `list(...)` is actually inefficient here: it requires finding all results, even though the first one would be sufficient.

Reply:
That said... I'm not sure I like the logic. It uses `checks.yml` - except when there's at least one `checks` field in the build file?

Reply:
DONE re list.
COMMENT ADDED re logic -- inlined checks (in build.yml) should override external checks.yml -- otherwise, you can't safely create libraries of checks.